drivers/video/video_stm32_venc.c

```diff
@@ -0,0 +1,1026 @@
+			EWL_HEAP_ALIGNED_FREE(inst->chunks[i]);
+			break;
+		}
+	}
+	info->virtualAddress = NULL;
+	info->busAddress = 0;
+	info->size = 0;
+
+	LOG_DBG("%s<\n", __func__);
+}
+
+i32 EWLReserveHw(const void *inst)
+{
+	LOG_DBG("%s>\n", __func__);
+	__ASSERT_NO_MSG(inst != NULL);
+	LOG_DBG("%s<\n", __func__);
+	return EWL_OK;
+}
+
+void EWLReleaseHw(const void *inst)
+{
+	LOG_DBG("%s>\n", __func__);
+	__ASSERT_NO_MSG(inst != NULL);
+	LOG_DBG("%s<\n", __func__);
+}
+
+void *EWLmalloc(u32 n)
+{
+	void * p = NULL;
+
+	LOG_DBG("%s> n=%d\n", __func__, n);
+
+	p = EWLcalloc(n, 1);
+
+#ifdef TRACE_EWL
+	mem_counter += n;
+	LOG_DBG("EWLmalloc %8d bytes --> %p, total : %d\n", n, p, mem_counter);
+#endif /* TRACE_EWL */
+
+	LOG_DBG("%s<\n", __func__);
+	return p;
+}
+
+void EWLfree(void *p)
+{
+	EWL_HEAP_ALIGNED_FREE(p);
+}
+
+void *EWLcalloc(u32 n, u32 s)
+{
+	void *p;
+
+	LOG_DBG("%s> n=%d s=%d\n", __func__, n, s);
+
+	p = EWL_HEAP_ALIGNED_ALLOC(n * s);
+	if (!p) {
+		LOG_ERR("alloc failed for size=%d\n", n * s);
+		return NULL;
+	}
+
+	memset(p, 0, n * s);
+
+	LOG_DBG("%s< %p\n", __func__, p);
+	return p;
+}
+
+void *EWLmemcpy(void *d, const void *s, u32 n)
+{
+	return memcpy(d, s, (size_t)n);
+}
+
+void *EWLmemset(void *d, i32 c, u32 n)
+{
+	return memset(d, c, (size_t)n);
+}
+
+int EWLmemcmp(const void *s1, const void *s2, u32 n)
+{
+	return memcmp((const uint8_t *) s1, (const uint8_t *) s2, (size_t)n);
+}
+
+i32 EWLWaitHwRdy(const void *inst, u32 *slicesReady)
+{
+	LOG_DBG("%s>\n", __func__);
+
+	/* Check invalid parameters */
+	if (!inst)
+		return EWL_HW_WAIT_ERROR;
+
+#ifdef EWL_USE_POLLING_SYNC
+	u32 ret = EWL_HW_WAIT_TIMEOUT;
+	volatile u32 irq_stats;
+	u32 prevSlicesReady = 0;
+	k_timepoint_t timeout = sys_timepoint_calc(K_MSEC(EWL_TIMEOUT));
+	uint32_t start = sys_clock_tick_get_32();
+
+	/* check how to clear IRQ flags for VENC */
+	u32 clrByWrite1 = (EWLReadReg(inst, BASE_HWFuse2) & HWCFGIrqClearSupport);
+	do {
+		irq_stats = VENC_REG(1UL);
+		/* get the number of completed slices from ASIC registers. */
+		if (slicesReady)
+			*slicesReady = (VENC_REG(21UL) >> 16) & 0xFFUL;
+
+		LOG_DBG("EWLWaitHw: IRQ stat = %08x\n", irq_stats);
+		u32 hw_handshake_status = READ_BIT(VENC_REG(BASE_HEncInstantInput >> 2U), (1U << 29U));
+
+		/* ignore the irq status of input line buffer in hw handshake mode */
+		if ((irq_stats == ASIC_STATUS_LINE_BUFFER_DONE) && (hw_handshake_status != 0UL)) {
+			VENC_REG(1U) = (1<<9);
+			continue;
+		}
+
+		if ((irq_stats & ASIC_STATUS_ALL) != 0UL) {
+			/* clear IRQ and slice ready status */
+			u32 clr_stats;
+			irq_stats &= (~(ASIC_STATUS_SLICE_READY | ASIC_IRQ_LINE));
+
+			if (clrByWrite1 != 0UL)
+				clr_stats = ASIC_STATUS_SLICE_READY | ASIC_IRQ_LINE;
+			else
+				clr_stats = irq_stats;
+
+			EWLWriteReg(inst, 0x04, clr_stats);
+			ret = EWL_OK;
+			break;
+		}
+
+		if (slicesReady)
+			if (*slicesReady > prevSlicesReady) {
+				ret = EWL_OK;
+				break;
+			}
+
```

Comment:
```suggestion
		if (slicesReady) {
			if (*slicesReady > prevSlicesReady) {
				ret = EWL_OK;
				break;
			}
		}
```
or combine in same condition: `if (slicesReady != NULL && *slicesReady > prevSlicesReady)`